### PR TITLE
ISPN-7290 Document Data Container Changes

### PR DESCRIPTION
**Pull request by wburns**
* Changed eviction section to have data container
* Added in caffeine and off heap

https://issues.jboss.org/browse/ISPN-7290

### CODE REVIEW

File: documentation/src/main/asciidoc/user_guide/eviction.adoc [hunk 1]
```diff
@@ -23,29 +27,19 @@ attribute will probably take some tuning in each use case.
...
-
-.`MANUAL`
-This eviction strategy is identical to NONE, in that it disables automatic eviction altogether, but signals the intention that the user wants to evict entries manually. The effect is to disable misleading warning validation messages.
+Eviction is handled by link:https://github.com/ben-manes/caffeine[Caffeine] utilizing
+the TinyLFU with an additional admission window.  This was chosen as provides high hit
```
Comment (tristantarrant) [inline documentation/src/main/asciidoc/user_guide/eviction.adoc]: /TinyLFU/TinyLFU algorithm/

File: documentation/src/main/asciidoc/user_guide/eviction.adoc [hunk 2]
```diff
@@ -61,27 +55,43 @@ the configured `size`.  This type only works with primitive wrapper, String
...
 
+==== Storage type
+
+Infinispan allows the user to configure in what form their data is stored.  Each
+form supports the same features of Infinispan, however evicttion can be limited
+for some forms.  There are currently three storage formats that Infinispan
+provides, they are:
+
+.`OBJECT`
+Stores the keys and values as is provided from the user.  Only `COUNT` eviction
```
Comment (tristantarrant) [inline documentation/src/main/asciidoc/user_guide/eviction.adoc]: Stores the keys and values as objects. 

I would mention that this means on-heap

File: documentation/src/main/asciidoc/user_guide/eviction.adoc [hunk 3]
```diff
@@ -61,27 +55,43 @@ the configured `size`.  This type only works with primitive wrapper, String
...
+type is supported.
+
+.`BINARY`
+Stores the keys and values as a byte[] under the scenes.  Both `COUNT` and
+`MEMORY` eviction types are supported.
+
+.`OFF-HEAP`
+Stores the keys and values in native memory outside of the
+perview of the Garbage Collector as a byte[].  Both `COUNT` and
```
Comment (tristantarrant) [inline documentation/src/main/asciidoc/user_guide/eviction.adoc]: purview

File: documentation/src/main/asciidoc/user_guide/eviction.adoc [hunk 4]
```diff
@@ -61,27 +55,43 @@ the configured `size`.  This type only works with primitive wrapper, String
...
+`MEMORY` eviction types are supported.
+
 ==== More defaults
 
-By default when no `<eviction />` element is specified, no eviction takes place.
+By default when no `<memory />` element is specified, no eviction takes place and
+`OBJECT` storage type is used.
 
-In case there is an eviction element, this table describes behaviour of eviction
+In case there is an memory element, this table describes behaviour of eviction
```
Comment (tristantarrant) [inline documentation/src/main/asciidoc/user_guide/eviction.adoc]: describes *the* behaviour

**Comment by wburns** [comment]
Fixed the comments, thanks @tristantarrant !

**Pull request merged**